src/librustc_span/lib.rs

```diff
@@ -1104,14 +1125,19 @@ impl SourceFile {
             crate_of_origin: 0,
             src: Some(Lrc::new(src)),
             src_hash,
-            external_src: Lock::new(ExternalSource::Unneeded),
+            external_src: Lock::new(ExternalSource {
+                kind: ExternalSourceKind::Unneeded,
+                original_start_pos: BytePos(0),
```

Comment:
These values should never get used - I want something that looks 'obviously wrong' in case they do.

Alternatively, we could refactor `ExternalSource` into an enum that only stores `original_start_pos`/`original_end_pos` in the variants that need them.